packages/esm-commons-lib/src/utils/createOHRIGroupedLink.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo, useRef } from 'react';
+import classNames from 'classnames';
+import { BrowserRouter, useLocation } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { ConfigurableLink } from '@openmrs/esm-framework';
+import { SideNavMenu } from '@carbon/react';
+
+interface DashboardLinkConfig {
+  name: string;
+  title: string;
+  isFolder: boolean;
+  folderTitle: string;
+  folderIcon: React.ComponentType<any>;
+  childLinks?: Array<any>;
+  isHidden?: boolean;
+}
+
+function DashboardExtension({ dashboardLinkConfig }: { dashboardLinkConfig: DashboardLinkConfig }) {
+  const { t } = useTranslation();
+  const { name, title, isFolder, folderTitle, folderIcon, childLinks, isHidden } = dashboardLinkConfig;
+  const location = useLocation();
+  const spaBasePath = `${window.spaBase}/home`;
+
+  const navLink = useMemo(() => {
+    const pathArray = location.pathname.split('/home');
+    const lastElement = pathArray[pathArray.length - 1];
+    return decodeURIComponent(lastElement);
+  }, [location.pathname]);
+
+  if (isHidden) {
+    return;
+  }
+
+  if (isFolder) {
```

Comment:
This is an implementation that other people can take advantage of. Ampath are interested in this as well.

Reply:
i can do that in a follow up PR